machado/loaders/assay.py

```diff
@@ -0,0 +1,123 @@
+from machado.models import Cv, Cvterm, Project
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.utils import IntegrityError
+from typing import Dict, List, Set, Union
+from datetime import datetime
+
+class AssayLoader(object):
+    """Load Assay."""
+
+    help = 'Load assay record.'
+
+    def __init__(self,
+                 db: str=None) -> None:
+        """Execute the init function."""
+        # get database for assay (e.g.: "SRA" - from NCBI)
+        try:
+            self.db, created = Db.objects.get_or_create(name=db)
+        except IntegrityError as e:
+            self.db = None
+
+        # will not use arraydesign nor operator
+        self.db_null, created = Db.objects.get_or_create(name='null')
+        self.dbxref_null, created = Dbxref.objects.get_or_create(
+            db=self.db_null, accession='null')
+        self.cv_null, created = Cv.objects.get_or_create(name='null')
+        self.cvterm_null, created = Cvterm.objects.get_or_create(
+            cv=self.cv_null,
+            name='null',
+            definition='',
+            dbxref=self.dbxref_null,
+            is_obsolete=0,
+            is_relationshiptype=0)
+        self.contact_null, created = Contact.objects.get_or_create(
+            name='null',
+            type_id=self.cvterm_null.cvterm_id)
+        self.arraydesign_null, created = Arraydesign.objects.get_or_create(
+            manufacturer_id=self.contact_null.contact_id,
+            platformtype_id=self.cvterm_null.cvterm_id,
+            name="Null")
+
+    def store_assay(self,
+                    name: str=None,
+                    acc: str=None,
+                    assaydate: str=None,
+                    description: str=None) -> None:
+        """Store assay."""
+
+        if assaydate:
+            # format date mandatory, e.g.: Oct-16-2016)
+            # in settings.py set USE_TZ = False
+            date_format = '%b-%d-%Y'
+            assaydate = datetime.strptime(assaydate, date_format)
+        # for example, acc is the "SRRxxxx" experiment accession from SRA
+        try:
+            self.dbxref, created = Dbxref.objects.get_or_create(
+                                                       accession=acc,
+                                                       db=self.db)
+        except IntegrityError as e:
+            self.dbxref = None
+            #create assay object
+        try:
+            self.assay, created = Assay.objects.get_or_create(
+                            arraydesign=self.arraydesign_null,
+                            operator_id=self.contact_null.contact_id,
+                            assaydate=assaydate,
+                            dbxref=self.dbxref,
+                            name=name,
+                            description=description)
+        except IntegrityError as e:
+            raise ImportingError(e)
+
+    def store_assay_project(self,
+                            project: Union[str, Project]):
```

Comment:
I think this function should require the parameter assay as well, rather than get it from self.

"Explicit is better than implicit."
- PEP 20 -- The Zen of Python